### PR TITLE
vim-patch: update runtime files

### PR DESCRIPTION
**Pull request by clason**
- **vim-patch:9.1.0591: filetype: `*.wl` files are not recognized**
- **vim-patch:9.1.0592: runtime: filetype: Mediawiki files are not recognized**
- **vim-patch:6a54dcb: runtime(gdscript): add recommended indent options to ftplugin**
- **vim-patch:3e07d5a: runtime(go): add recommended indent options to ftplugin**
- **vim-patch:babea52: runtime(gomod): add recommended indent options to ftplugin**
- **vim-patch:9.1.0593: filetype: Asymptote files are not recognized**
- **vim-patch:3698fbb: runtime(tsv): include simple syntax plugin**

### CODE REVIEW

File: runtime/doc/syntax.txt [hunk 1]
```diff
@@ -1846,7 +1865,16 @@ Empty `*.m` files will automatically be presumed to be Matlab files unless you
...
+<
+MEDIAWIKI					*ft-mediawiki-syntax*
+
+Be default, syntax highlighting includes basic HTML tags like style and
+headers |html.vim|. For strict Mediawiki syntax highlighting: >
+	let g:html_no_rendering = 1
+If HTML highlighting is desired, terminal-based text formatting such as bold
+and italic is possible by: >
+	let g:html_style_rendering = 1
+<
```
Comment (zeertzjq) [inline runtime/doc/syntax.txt:1877]:
```suggestion

MEDIAWIKI					*ft-mediawiki-syntax*

Be default, syntax highlighting includes basic HTML tags like style and
headers |html.vim|. For strict Mediawiki syntax highlighting: >

	let g:html_no_rendering = 1

If HTML highlighting is desired, terminal-based text formatting such as bold
and italic is possible by: >

	let g:html_style_rendering = 1
```

File: runtime/ftplugin/gdscript.vim [hunk 2]
```diff
@@ -26,6 +26,11 @@ setlocal foldignore=
 setlocal foldexpr=s:GDScriptFoldLevel()
 
 
+if get(g:, 'gdscript_recommended_style', 1)
+    let setlocal noexpandtab tabstop=4 softtabstop=0 shiftwidth=0
```
Comment (zeertzjq) [inline runtime/ftplugin/gdscript.vim:30]:
```suggestion
    setlocal noexpandtab tabstop=4 softtabstop=0 shiftwidth=0
```

File: runtime/ftplugin/gdscript.vim [hunk 3]
```diff
@@ -26,6 +26,11 @@ setlocal foldignore=
 setlocal foldexpr=s:GDScriptFoldLevel()
```
Comment (zeertzjq) [inline runtime/ftplugin/gdscript.vim:28]:
```suggestion
```

File: runtime/ftplugin/gdscript.vim [hunk 4]
```diff
@@ -26,6 +26,11 @@ setlocal foldignore=
...
 
 
+if get(g:, 'gdscript_recommended_style', 1)
+    let setlocal noexpandtab tabstop=4 softtabstop=0 shiftwidth=0
+    let b:undo_ftplugin ..= ' | setlocal expandtab< tabstop< softtabstop< shiftwidth<'
+endif
+
```
Comment (zeertzjq) [inline runtime/ftplugin/gdscript.vim:33]:
```suggestion


```

File: runtime/doc/syntax.txt (hunk 1 above)
Comment (clason) [reply runtime/doc/syntax.txt:1877]: I don't think we should accept broken syntax any longer which can mess up header detection. Our primary "record of truth" for valid vimdoc syntax is now our tree-sitter parser, not upstream's Pirates' Code.

Comment (zeertzjq) [reply runtime/doc/syntax.txt]: What messes up header detection? Upstream docs looks completely fine with treesitter.

Comment (clason) [reply runtime/doc/syntax.txt]: Yes, in this case. Other cases have _not_ been fine and I had to correct them manually to get the docs check to pass.

File: runtime/doc/syntax.txt [hunk 5]
```diff
@@ -1846,7 +1865,15 @@ Empty `*.m` files will automatically be presumed to be Matlab files unless you
 have the following in your vimrc: >
 
 	let filetype_m = "mma"
-
+
```
Comment (zeertzjq) [inline runtime/doc/syntax.txt:1868]:
```suggestion

```

File: runtime/doc/syntax.txt [hunk 6]
```diff
@@ -1846,7 +1865,15 @@ Empty `*.m` files will automatically be presumed to be Matlab files unless you
 have the following in your vimrc: >
 
 	let filetype_m = "mma"
-
+ 
+MEDIAWIKI					*ft-mediawiki-syntax*
+
+Be default, syntax highlighting includes basic HTML tags like style and
+headers |html.vim|. For strict Mediawiki syntax highlighting: >
+	let g:html_no_rendering = 1
```
Comment (zeertzjq) [inline runtime/doc/syntax.txt:1873]:
```suggestion

	let g:html_no_rendering = 1

```

File: runtime/doc/syntax.txt [hunk 7]
```diff
@@ -1846,7 +1865,15 @@ Empty `*.m` files will automatically be presumed to be Matlab files unless you
...
+If HTML highlighting is desired, terminal-based text formatting such as bold
+and italic is possible by: >
+	let g:html_style_rendering = 1
```
Comment (zeertzjq) [inline runtime/doc/syntax.txt:1876]:
```suggestion

	let g:html_style_rendering = 1
```

File: runtime/doc/syntax.txt [hunk 8]
```diff
@@ -1847,6 +1866,17 @@ have the following in your vimrc: >
 
 	let filetype_m = "mma"
 
+MEDIAWIKI					*ft-mediawiki-syntax*
+
+Be default, syntax highlighting includes basic HTML tags like style and
```
Comment (clason) [inline runtime/doc/syntax.txt:1871]: Typo, but I'm leaving this untouched.

**Review by zeertzjq — APPROVED** [review]

**Pull request merged**